nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -729,8 +757,40 @@ define([
             updateState();
         }
 
+        /**
+         * @param {string} action
+         * @returns
+         */
         function runAction(action) {
-            alert(action);
+            if (readOnly) {
+                console.warn('ignoring attempted action in readonly mode');
+                return;
+            }
+            switch (action) {
+                case 'runApp':
+                    cell.execute();
```

Comment:
`runApp` is the only one implemented right now. Just tells the Jupyter cell to run itself.